### PR TITLE
Essence high tier filter added to tft/bulk-tool, Next and Previous buttons updated.

### PR DESCRIPTION
**Pull request by dwoolley**
Added two sub filters "essence - deafening+" and "essence - shrieking+" onto the tft/bulk-tool page.
Next and Previous buttons will now be disabled when at the start and end of some lists.

### CODE REVIEW

File: src/components/item-table/item-table-body.tsx [hunk 1]
```diff
@@ -21,12 +21,14 @@ export default function ItemTableBody({
...
   setSearchUserInput,
   disableTotalValueRow,
+  setHasMore,
 }: {
   itemGroupSearch: StashSnapshotItemGroupSummarySearch;
   league: string;
   searchUserInput: ItemSearchUserInput;
   setSearchUserInput: Dispatch<SetStateAction<ItemSearchUserInput>>;
   disableTotalValueRow?: boolean;
+  setHasMore: Function;
```
Comment (JVictorV) [inline src/components/item-table/item-table-body.tsx:31]:
```suggestion
  setHasMore: Dispatch<SetStateAction<boolean>>;
```

File: src/components/item-table/item-table-body.tsx [hunk 2]
```diff
@@ -63,6 +65,10 @@ export default function ItemTableBody({
       variables: { search: itemGroupSearch },
       skip: !itemGroupSearch,
       onCompleted(data) {
+        try {
+          setHasMore(data.stashSnapshotItemGroupSummaries.hasMore);
+        } catch { }
```
Comment (JVictorV) [inline src/components/item-table/item-table-body.tsx:70]: Why do we need this try catch here?

**Review by JVictorV — CHANGES_REQUESTED** [review]
Great job on the implementation! Based on my testing, it seems to be working well. There are just a few points that need to be reviewed before we can merge it.

File: src/components/item-table/item-table-body.tsx (hunk 2 above)
Comment (dwoolley) [reply src/components/item-table/item-table-body.tsx:70]: I had set more as just a function, that was being passed in and wasn't sure if all sources would have to send the function. Now that it's a Dispatch<SetStateAction<boolean>>, I assume the type is going to be more accurate and my concerns for safety are unwarranted. I'll remove the try catch

**Review by JVictorV — APPROVED** [review]
Looks good to me! thanks for contributing

**Pull request closed without merging**